interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -1,19 +1,31 @@
 angular.module('adage.gene.searchMany', [
+  'ui.router',
   'adage.gene.resource',
   'adage.gene.utils',
   'adage.gene.selected',
+  'adage.utils',
   'adage.mlmodel.components'
 ])
 
 .config(function($stateProvider) {
   $stateProvider
     .state('gene_search', {
-      url: '/gene_search',
+      url: '/gene_search?mlmodel',
       views: {
         'main': {
           templateUrl: 'gene/gene-network.tpl.html',
-          controller: ['UserFactory', function(UserFactory) {
+          controller: ['$stateParams', 'MlModelTracker', 'UserFactory',
+          function($stateParams, MlModelTracker, UserFactory) {
             var self = this;
+            self.isValidModel = false;
+            // Do nothing if mlmodel in URL is falsey. The error will be taken
+            // care of by "<ml-model-validator>" component.
+            if (!$stateParams.mlmodel) {
+              return;
+            }
+
+            self.modelInUrl = $stateParams.mlmodel;
+            self.selectedMlModel = MlModelTracker;
```

Comment:
This is a cool design, by the way. I like that the service itself is an object with properties that yield the relevant MLModel details.